crates/router/src/core/payment_methods.rs

```diff
@@ -1825,6 +1821,72 @@ pub async fn update_payment_method(
     Ok(services::ApplicationResponse::Json(response))
 }
 
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+#[instrument(skip_all)]
+pub async fn delete_payment_method(
+    state: SessionState,
+    pm_id: api::PaymentMethodId,
+    key_store: domain::MerchantKeyStore,
+    merchant_account: domain::MerchantAccount,
+) -> RouterResponse<api::PaymentMethodDeleteResponse> {
+    let db = state.store.as_ref();
+    let key_manager_state = &(&state).into();
+
+    let pm_id = id_type::GlobalPaymentMethodId::generate_from_string(pm_id.payment_method_id)
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Unable to generate GlobalPaymentMethodId")?;
+
+    let payment_method = db
+        .find_payment_method(
+            &((&state).into()),
+            &key_store,
+            &pm_id,
+            merchant_account.storage_scheme,
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::PaymentMethodNotFound)?;
+
+    let locker_id = payment_method.locker_id.clone();
+
+    let _customer = db
+        .find_customer_by_merchant_reference_id_merchant_id(
```

Comment:
```suggestion
        .find_customer(
```
we should find customer by customer global id